pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -3737,11 +3738,13 @@ private Topic getTopicReference(TopicName topicName) {
                     .orElseThrow(() -> topicNotFoundReason(topicName));
         } catch (RestException e) {
             throw e;
+        } catch (TimeoutException e){
```

Comment:
It looks unnecessary.

Reply:
Should I move it to ``Exception`` catch and then ``instanceof`` ? I mean if I don't return timeout. I can't tell if we're getting NPE.

Reply:
You can check has `Exception` and should be `java.util.concurrent.TimeoutException` instance.

Reply:
We can only get server internal errors without any timeout text, if we do not handle ``Timeoutexception`` separately.

![image](https://user-images.githubusercontent.com/74767115/148650283-dab5139c-afda-489d-971d-fe3bb7977960.png)

Reply:
@mattisonchao You can check the HTTP response body, the HTTP detail:
```
NettyResponse {
	statusCode=500
	headers=
		Date: Sat, 08 Jan 2022 16:07:22 GMT
		broker-address: localhost
		Content-Type: text/plain
		Server: Jetty(9.4.43.v20210629)
		content-length: 2236
	body=

 --- An unexpected error occurred in the server ---

Message: null

Stacktrace:

java.util.concurrent.TimeoutException
	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1886)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2021)
	at org.apache.pulsar.broker.admin.impl.PersistentTopicsBase.getTopicReference(PersistentTopicsBase.java:3737)
	at org.apache.pulsar.broker.admin.impl.PersistentTopicsBase.internalResetCursorForNonPartitionedTopic(PersistentTopicsBase.java:2012)
	at org.apache.pulsar.broker.admin.impl.PersistentTopicsBase.lambda$internalResetCursor$93(PersistentTopicsBase.java:1992)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire$$$capture(CompletableFuture.java:714)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073)
	at org.apache.pulsar.metadata.impl.ZKMetadataStore.handleGetResult(ZKMetadataStore.java:227)
	at org.apache.pulsar.metadata.impl.ZKMetadataStore.lambda$batchOperation$6(ZKMetadataStore.java:177)
	at org.apache.zookeeper.MockZooKeeper.multi(MockZooKeeper.java:982)
	at org.apache.pulsar.metadata.impl.ZKMetadataStore.batchOperation(ZKMetadataStore.java:148)
	at org.apache.pulsar.metadata.impl.batching.AbstractBatchedMetadataStore.flush(AbstractBatchedMetadataStore.java:90)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.runAndReset$$$capture(FutureTask.java:305)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)

}
```

Reply:
fixed